contrib/repro_test.bzl

```diff
@@ -0,0 +1,247 @@
+    _extract.implementation(
+        ctx,
+        name = "build_image_and_extract",
+        commands = commands,
+        extract_file = extract_path,
+        image = image_tar,
+        docker_run_flags = docker_run_flags,
+        output_file = img1_outs,
+        script_file = ctx.actions.declare_file(name + ".build1"),
+    )
+
+    # Build and extract 'image' again in a different container to try and
+    # repro previously built 'image'.
+    _extract.implementation(
+        ctx,
+        name = "build_image_and_extract_repro",
+        commands = commands,
+        extract_file = extract_path,
+        image = image_tar,
+        docker_run_flags = docker_run_flags,
+        output_file = img2_outs,
+        script_file = ctx.actions.declare_file(name + ".build2"),
+    )
+
+    # Expand template to run the image comparison test.
+    type_args = ["--type=" + type_arg for type_arg in ctx.attr.container_diff_args]
+    diff_tool_exec = ctx.executable._container_diff_tool
+    ctx.actions.expand_template(
+        template = ctx.file._test_tpl,
+        substitutions = {
+            "%{container_diff_args}": " ".join(type_args),
+            "%{container_diff_tool}": diff_tool_exec.short_path,
+            "%{img1_path}": img1_outs.short_path,
+            "%{img2_path}": img2_outs.short_path,
+            "%{img_name}": img_label.name,
+        },
+        output = ctx.outputs.test_script,
+        is_executable = True,
+    )
+
+    runfiles = ctx.runfiles(files = [img1_outs, img2_outs, diff_tool_exec])
+
+    return [
+        DefaultInfo(
+            executable = ctx.outputs.test_script,
+            runfiles = runfiles,
+        ),
+    ]
+
+container_repro_test = rule(
+    attrs = dicts.add(_container.image.attrs, {
+        "base": attr.label(
+            default = "@bazel_latest//image",
+            allow_rules = [
+                "container_image_",
+                "container_import",
+                "toolchain_container_",
+            ],
+            doc = "An image target compatible with the `base` attribute of " +
+                  "the container_image rule to build and reproduce the " +
+                  "'image' in. This image must have Bazel and docker " +
+                  "installed and available in the PATH.",
+        ),
+        "container_diff_args": attr.string_list(
+            default = ["file", "size", "apt", "pip"],
+            doc = "List of the --type flag values to pass to the container_diff tool. " +
+                  "Check https://github.com/GoogleContainerTools/container-diff#container-diff " +
+                  "for more info.",
+        ),
+        "image": attr.label(
+            allow_rules = [
+                "container_image_",
+                "toolchain_container_",
+            ],
+            mandatory = True,
+            doc = "A container_image target to test for reproducibility.",
+        ),
+        "workspace_file": attr.label(
+            allow_single_file = True,
```

Comment:
lets try to do 
`allow_single_file = ["WORKSPACE"],`

Reply:
Done